attribute_set_completeness/models/attribute_set_owner_mixin.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class AttributeSetOwnerMixin(models.AbstractModel):
+
+    _inherit = "attribute.set.owner.mixin"
+
+    completion_rate = fields.Float(compute="_compute_completion_rate", readonly=True)
+    completion_state = fields.Selection(
+        selection=[("complete", "Complete"), ("not_complete", "Not complete")],
+        compute="_compute_completion_rate",
+        search="_search_complete_state",
+    )
+
+    @api.multi
+    def _compute_completion_rate(self):
+        for record in self:
+            attribute_set_id = record.attribute_set_id
+            completion_config = attribute_set_id.attribute_set_completeness_ids
+            if completion_config:
+                completion_rate = 0.0
+                for criteria in completion_config:
+                    field_name = criteria.field_id.name
+                    if record[field_name]:
+                        completion_rate += criteria.completion_rate
+                record.completion_rate = completion_rate
+                if completion_rate < 1:
+                    record.completion_state = "not_complete"
+                else:
+                    record.completion_state = "complete"
+            else:
+                record.completion_rate = False
+                record.completion_state = False
+
+    @api.multi
+    def _search_complete_state(self, operator, value):
+        products = self.search([("attribute_set_id", "!=", False)])
```

Comment:
I would move that search before 'if operator == "=":' to avoid excessive searches.